src/item.cpp

```diff
@@ -3661,6 +3661,10 @@ bool item::can_contain( const itype &tp ) const
 
 bool item::spill_contents( Character &c )
 {
+    if( is_container_empty() || !is_container() ) {
```

Comment:
`is_container_empty()` checks just `contents.empty()`, meaning speed difference would be negligible. I'll change for readability, though.